File: OCH_Tao/ABC/B/301.py
N = int(input())
A = list(map(int, input().split()))
L = []
for i in range(N-1):
  if A[i] < A[i+1]:
    L.extend(list(range(A[i], A[i+1])))
  else:
    L.extend(list(range(A[i], A[i+1], -1)))
L.append(A[-1])
print(*L)

File: OCH_Tao/ABC/B/301.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const N = parseInt(input[0], 10);
const A = input[1].split(" ").map(x => parseInt(x, 10));
let L = [];
for (let i = 0; i < N - 1; i++) {
  if (A[i] < A[i + 1]) {
    for (let j = A[i]; j < A[i + 1]; j++) {
      L.push(j);
    }
  } else {
    for (let j = A[i]; j > A[i + 1]; j--) {
      L.push(j);
    }
  }
}
L.push(A[N - 1]);
console.log(...L);